cmd/clusterctl/client/client_test.go

```diff
@@ -216,10 +221,19 @@ func (p *fakeCertManagerClient) EnsureLatestVersion() error {
 	return nil
 }
 
+func (p *fakeCertManagerClient) PlanUpgrade() (cluster.CertManagerUpgradePlan, error) {
+	return p.certManagerPlan, nil
+}
+
 func (p *fakeCertManagerClient) Images() ([]string, error) {
 	return p.images, p.imagesError
 }
 
+func (p *fakeCertManagerClient) WithCertManagerPlan(plan CertManagerUpgradePlan) *fakeCertManagerClient {
```

Comment:
Might be it will be more expressive to set the installed version and the embedded version instead of passing a plan

Reply:
I passed the plan in directly instead of just the version numbers so I could be a little "future-proof". Also, I would need to also pass in `ShouldUpgrade` boolean else it will be always `false`. So instead of passing in the individual properties, I thought I'd just pass in the plan directly.